# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: browser/base/content/test/contextMenu/browser_strip_on_share_nested_link.js
# Commit: 44bca4706eb2
# Full Hash: 44bca4706eb261f8e3bcbcb44a6de7cd7d7c3874
# Author: abhishekmadan30 <148475204+abhishekmadan30@users.noreply.github.com>
# Date: 2024-04-10 09:45:41
# Description:
#   Bug 1888734 - Added null check to nested strip on share function. r=pbz,anti-tracking-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206559
# ==============================================================================

diff -r b8a581227ff5 -r 44bca4706eb2 browser/base/content/test/contextMenu/browser_strip_on_share_nested_link.js
--- a/browser/base/content/test/contextMenu/browser_strip_on_share_nested_link.js	Tue Apr 09 17:07:53 2024 +0000
+++ b/browser/base/content/test/contextMenu/browser_strip_on_share_nested_link.js	Tue Apr 09 17:10:53 2024 +0000
@@ -84,6 +84,16 @@
   });
 });
 
+// Testing functionality with no https pages
+add_task(async function testNonHTTPsPages() {
+  let validUrl = "https://www.example.com/?test_2=1234&test=about%3A%3Aconfig";
+  let shortenedUrl = "https://www.example.com/?test=about%3A%3Aconfig";
+  await testStripOnShare({
+    originalURI: validUrl,
+    strippedURI: shortenedUrl,
+  });
+});
+
 /**
  * Opens a new tab, opens the context menu and checks that the strip-on-share menu item is visible.
  * Checks that the stripped version of the url is copied to the clipboard.